### PR TITLE
Agcol - verify and timeout 

### PR DESCRIPTION
**Pull request by AGCol11**

### CODE REVIEW

**Comment by CatusKing** [comment]
You need to list the commands you worked on.

**Comment by CatusKing** [comment]
### Kick and Ban
- Needs to log in a log channel
- Response to reply can be short like "Banned CatusKing"
- Commands should only be visible to people with ban perms. This means changing the slash command builder so it shows only to people with certain perms. This is rather then checking perms.
- the kick function for the member object calls for a string as a reason not an object
### Timeout
- remove embeds but keep formatt
- remove timeout delete button
- we should probably add a cancel timeout feature
### Verify
- doesnt make sure file type is an image
- also should use both tag and ping to show who used it

**Comment by AGCol11** [comment]
fixed all noted

File: src/commands/Moderation/ban.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    .setName('ban')
+    .setDescription('Bans a user')
+    .addUserOption(option => option.setName('user').setDescription('Select the user you wish to ban!').setRequired(true))
+    .addStringOption(option => option.setName('reason').setDescription('Reason for banning this user').setRequired(true))
+    .setDefaultMemberPermissions(PermissionsBitField.Flags.BanMembers),
+
+
+    async execute(interaction, client) {
+        const user = interaction.options.getUser('user')
+        const reason = interaction.options.getString('reason') || 'No reason provided';
```
Comment (CatusKing) [inline src/commands/Moderation/ban.js:14]: If the command option is required you dont need an "or" for the definition.

File: src/commands/Moderation/kick.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    .setName('kick')
+    .setDescription('Kicks a user')
+    .addUserOption(option => option.setName('user').setDescription('Select the user you wish to kick!').setRequired(true))
+    .addStringOption(option => option.setName('reason').setDescription('Reason for kicking this user').setRequired(true))
+    .setDefaultMemberPermissions(PermissionsBitField.Flags.KickMembers),
+
+
+    async execute(interaction, client) {
+        const user = interaction.options.getUser('user')
+        const reason = interaction.options.getString('reason') || 'No reason provided';
```
Comment (CatusKing) [inline src/commands/Moderation/kick.js]: If the command option is required you dont need an "or" for the definition.

File: src/commands/Moderation/timeout.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                    { name: '3 Hours', value: '10800' },
+                    { name: '5 Hours', value: '18000' },
+                    { name: '10 Hours', value: '36000' },
+                    { name: '1 Day', value: '86400' },
+                    { name: '2 Days', value: '172800' },
+                    { name: '3 Days', value: '259200' },
+                    { name: '5 Days', value: '432000' },
+                    { name: '1 Week', value: '604800' },
+                ))
+        .addStringOption(option => option.setName('reason').setDescription('Reason for timing out the user')),
```
Comment (CatusKing) [inline src/commands/Moderation/timeout.js]: Please require the option and update code accordingly.

File: src/commands/Moderation/verify.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        ),
+        async execute(interaction) {
+            try {
+                const appliedRole = interaction.options.getString('position');
+        
+                const roleReply = await interaction.reply(`<@${interaction.user.id}>, you have requested verification for the role: **${appliedRole}**. Please provide proof of your application by sending an image.`);
+        
+                const proofCollector = interaction.channel.createMessageCollector({
+                    filter: msg => {
+                        if (msg.attachments.size > 0 && msg.attachments.every(attachment => attachment.contentType.startsWith('image'))) {
```
Comment (CatusKing) [inline src/commands/Moderation/verify.js]: Could you check the send to check if they are the interaction user.

**Review by CatusKing — CHANGES_REQUESTED** [review]
Just some simple things. Also sorry for updating your code with the verify. Realized I could use this feature instead.

File: src/commands/Moderation/verify.js (hunk 4 above)
Comment (CatusKing) [reply src/commands/Moderation/verify.js]: Phrased that horribly. When it recieves the callback you should confirm it's from the original sender.

File: src/commands/Moderation/timeout.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    async execute(interaction) {
+        const user = interaction.options.getUser('user');
+        const member = await interaction.guild.members.fetch(user.id);
+        const duration = interaction.options.getString('duration');
+
+
+
+        if(!interaction.member.permissions.has(PermissionsBitField.Flags.ModerateMembers)) return await interaction.reply({ content: 'You do not have permission to use this command', ephemeral: true});
```
Comment (CatusKing) [inline src/commands/Moderation/timeout.js]: Remember we have default permissions we can change in the command creation.

**Review by CatusKing — CHANGES_REQUESTED** [review]
Additional reminder on permission checking.

**Pull request merged**